### PR TITLE
CLOUDP-196391: seed replica set using the mongo driver instead of mogosh

### PR DESCRIPTION
**Pull request by tibulca**
_Jira ticket:_ CLOUDP-196391

### CODE REVIEW

File: internal/cli/atlas/deployments/setup.go [hunk 1]
```diff
@@ -187,30 +191,62 @@ func (opts *SetupOpts) configureMongod(ctx context.Context, keyFileContents stri
...
+	return opts.initReplicaSet(ctx)
+}
+
+func (opts *SetupOpts) initReplicaSet(ctx context.Context) error {
+	// connect to local deployment
+	const waitDuration = 60 * time.Second
+	ctxConnect, cancel := context.WithTimeout(ctx, waitDuration)
+	defer cancel()
+
+	client, errConnect := mongo.Connect(ctxConnect, mongoOpts.Client().ApplyURI(fmt.Sprintf("mongodb://localhost:%d/?directConnection=true", opts.Port)))
```
Comment (tibulca) [inline internal/cli/atlas/deployments/setup.go:203]: for testability, I'll create (as part of the current task) a separate PR to wrap the mongo client and RunCommand and InsertOne functions

File: internal/cli/atlas/deployments/setup.go [hunk 2]
```diff
@@ -253,14 +294,6 @@ func (opts *SetupOpts) configureMongot(ctx context.Context, keyFileContents stri
 	return err
 }
 
-func connString(port int) string {
-	return fmt.Sprintf("mongodb://localhost:%d/admin", port)
-}
-
-func (opts *SetupOpts) seed(port int, script string) error {
-	return mongosh.Exec(opts.debug, connString(port), "--eval", script)
```
Comment (tibulca) [inline internal/cli/atlas/deployments/setup.go:261]: changed with mongo driver as the setup was failing when mongosh was not installed

File: internal/cli/atlas/deployments/setup.go [hunk 3]
```diff
@@ -187,30 +191,67 @@ func (opts *SetupOpts) configureMongod(ctx context.Context, keyFileContents stri
...
+	defer cancel()
+
+	client, errConnect := mongo.Connect(ctxConnect, mongoOpts.Client().ApplyURI(connectionString))
+	if errConnect != nil {
+		return fmt.Errorf("%w: %w", errConnectFailed, errConnect)
+	}
+	if err := client.Ping(ctx, nil); err != nil {
+		return fmt.Errorf("%w: %w", errConnectFailed, err)
 	}
+	defer client.Disconnect(ctx) //nolint:errcheck
```
Comment (fmenezes) [inline internal/cli/atlas/deployments/setup.go:215]: _I think_ this should be before ping

File: internal/cli/atlas/deployments/setup.go [hunk 4]
```diff
@@ -187,30 +191,67 @@ func (opts *SetupOpts) configureMongod(ctx context.Context, keyFileContents stri
...
+			{
+				"_id":  0,
+				"host": fmt.Sprintf("%s:%d", opts.LocalMongodHostname(), internalMongodPort),
+				"horizons": bson.M{
+					"external": fmt.Sprintf("localhost:%d", opts.Port),
+				},
+			},
+		},
+	}
+	initiateRsCmd := bson.D{{Key: "replSetInitiate", Value: rsConfig}}
```
Comment (fmenezes) [inline internal/cli/atlas/deployments/setup.go:234]:
```suggestion
	initiateRsCmd := bson.D{{Key: "replSetInitiate", Value: bson.M{
		"_id":       replicaSetName,
		"version":   1,
		"configsvr": false,
		"members": []bson.M{
			{
				"_id":  0,
				"host": fmt.Sprintf("%s:%d", opts.LocalMongodHostname(), internalMongodPort),
				"horizons": bson.M{
					"external": fmt.Sprintf("localhost:%d", opts.Port),
				},
			},
		},
	}}}
```

**Review by fmenezes — APPROVED** [review]

File: internal/cli/atlas/deployments/setup.go (hunk 3 above)
Comment (fmenezes) [reply internal/cli/atlas/deployments/setup.go:215]: feel free to address on a followup

**Review by blva — APPROVED** [review]

**Pull request merged**